### ISSUE TITLE
Support linkbacks as in the blogosphere

### ISSUE THREAD

**Issue by archusr** [buckket/twtxt#109]
Consider two users _foo_ (https://foo.org/twtxt.txt) and _bar_ (https://bar.org/twtxt.txt).

When user _foo_ mentions user _bar_ in her tweet, the client could call the linkback url as defined in _bar_'s metadata (#22 #48), e.g. https://bar.org/twtxt/ping.php?url=https://foo.org/twtxt.txt&name=foo (name optional?)

ping.php would log the request to https://bar.org/twtxt/ping.txt (linkback log) as in:
`   2016-03-06T11:12:25+01:00	<@foo https://foo.org/twtxt.txt> mentioned you.`

As user _bar_ is subscribed to her linkback log she'd see the mention in her timeline.

Similar to pingback/trackback as in blogs, cf. https://en.wikipedia.org/wiki/Linkback. Sorry, if you already thought of all this.

Exemplary ping.php to write the linkback log ping.txt
```php
$line = date('Y-m-d\TH:i:sP')."\t"."<@$_GET['name'] $_GET['url']> mentioned you.";
$fh = fopen('ping.txt','a'); fwrite($fh, $line."\n"); fclose($fh);
```

**Comment by mdom** [comment]
If user foo is registered at a registry, user bar could just [query for mentions](http://twtxt.readthedocs.org/en/latest/user/registry.html#query-for-mentions) to see foo's tweet. Although i think it would be very nice to have a linkback plugin if a plugin system is ever added.

**Comment by archusr** [comment]
Ah, registries, thanks for pointing that out. It's a bit opposing the idea of decentralization, though.

**Comment by mdom** [comment]
The plan is to have multiple registries than can exchange their data. You would always be able to run your own registry.

**Comment by timofurrer** [comment]
I really like the idea of of the linkback - it must be optional though: if the linkback call fails it doesn't matter. 

> Ah, registries, thanks for pointing that out. It's a bit opposing the idea of decentralization, though.

I agree with that.

**Comment by mdom** [comment]
Yeah, you're right. I implemented basic support for Linkback in txtnix. Maybe we can discuss here what a twtxt client should send to a linkback URL. Is the URL enough? Anything else the endpoint would need? Should the endpoint check if user foo really mentioned user bar and only then write to the linkback log? Do you have an logback log we could play with archuser? But in the end it all hinges on our answer to #48.

**Comment by archusr** [comment]
Great, I like it a lot. I want to suggest:
LinkBack **must** send the _url_, **should** send the _name_, **may** send the _tweet_. (The tweet/text could then even be sent by mail to the user.)
The endpoint **should** check _foo_'s tweets for the mention to avoid spam, but that might be simply up to the implementation.

**Comment by DracoBlue** [comment]
:+1: 

Url of the twtxt.txt should be sifficient. Nick and real content of the tweet can be retrieved by the client ondemand.

So it is more like a ping ;)

**Comment by archusr** [comment]
You're right, URL is enough. Here's my initial [ping.php](https://github.com/archusr/twetwise/blob/master/ping/index.php) (working for me ;).

**Comment by Puddinglord** [comment]
I have a question for archusr, I am a college student working with a group that was going to work on this feature for a class, I see you have ping.php so would the link-back mention still be applicable to work on?

**Comment by archusr** [comment]
Hi @Puddinglord, thanks for your interest and support. The ping.php is just a simple/sample ping backend/receiver. At the moment, txtnix is the first and only twtxt client to support [linkbacks](https://github.com/mdom/txtnix/blob/master/lib/App/txtnix/Plugin/LinkBack.pm). I'm not sure if @buckket is interested in working code for twtxt -- which could be a nice project from my point of view, and I definately would be happy to see it :)

**Comment by Zegnat** [comment]
Just driving by and wanted to drop in and point people at [Webmention](https://webmention.net/), the latest W3C Recommendation for “a simple way to notify any URL when you mention it on your site”.

The syntax is basically as described by @archusr in https://github.com/buckket/twtxt/issues/109#issuecomment-196016659, except the parameters are called `source` and `target`.

**Comment by kixiQu** [comment]
The change would be that the webmention endpoint would have to be discoverable. 

> The sender must fetch the target URL (and follow redirects [FETCH]) and check for an HTTP Link header [RFC5988] with a rel value of webmention. If the content type of the document is HTML, then the sender must look for an HTML <link> and <a> element with a rel value of webmention. If more than one of these is present, the first HTTP Link header takes precedence, followed by the first <link> or <a> element in document order. Senders must support all three options and fall back in this order.

[webmention spec](https://www.w3.org/TR/webmention/) 

Adding a Link header is possible for anyone who's willing to fiddle with their web server, but isn't straightforward otherwise.

**Comment by zzo38** [comment]
If you want to reference a specific message (e.g. to reply to a question that it asks), then how are you supposed to know which message you are referencing? It seems to me that doing that would be more useful than merely a mention of the user name, which isn't very helpful in my opinion.
